### PR TITLE
Adjust collections of Tracker clusters to be saved for Phase-2 ALCARECO

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

This is a follow-up of PR https://github.com/cms-sw/cmssw/pull/39858.
After the backport of that PR (https://github.com/cms-sw/cmssw/pull/39898) was integrated, it was possible to test the production of RelVal samples (see e.g. on DAS: [/RelValMinBias_14TeV/CMSSW_12_5_2-TkAlMinBias-125X_mcRun4_realistic_v3_2026D88PU-v1/ALCARECO](https://cmsweb.cern.ch/das/request?view=list&limit=50&instance=prod%2Fglobal&input=dataset%3D/RelValMinBias_14TeV/CMSSW_12_5_2-TkAlMinBias-125X_mcRun4_realistic_v3_2026D88PU-v1/ALCARECO)).
When trying to run over it, it became clear that is not possible to refit tracks, due to the missing event products (the Phase-2 Outer Tracker clusters) that are necessary for CPE re-computation, leading to this error:

```
Begin processing the 1st record. Run 1, Event 301, LumiSection 4 on stream 6 at 13-Dec-2022 12:50:01.219 CET
----- Begin Fatal Exception 13-Dec-2022 12:50:57 CET-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 1 lumi: 4 event: 306 stream: 2
   [1] Running path 'p2'
   [2] Calling method for module TrackRefitter/'FinalTrackRefitter'
Exception Message:
RefCore: A request to resolve a reference to a product of type 'edmNew::DetSetVector<Phase2TrackerCluster1D>' with ProductID '4:449'
can not be satisfied because the product cannot be found.
Probably the branch containing the product is not stored in the input file.
   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
```

This PR adds the minimal amount of changes in the `CommonTools/RecoAlgos` package in order to be able to retain the Phase-2 OT clusters after running the `AlignmentTrackSelector` module that relies on it:

https://github.com/cms-sw/cmssw/blob/52c0dacc1af916e537cb419a39ec5845ae5087bf/Alignment/CommonAlignmentProducer/plugins/AlignmentTrackSelectorModule.cc#L13-L17

This is done in commit f5efa50bcfb2ebaf352f34992f59a3cb8b866a51.
Secondly the `outputCommands` of all the Tracker Alignment ALCARECO producers is modified in order to retain `Phase2TrackerCluster1DedmNewDetSetVector` in the output - in case the production is run for Phase-2 - while removing `SiStripClusteredmNewDetSetVector` (since there is no SiStrip detector in phase-2). I profit of that to re-adjust the commands for event contents to cope with no SCAL in Run3 that were leftover from https://github.com/cms-sw/cmssw/pull/38415 and https://github.com/cms-sw/cmssw/pull/38590. 
This is done in commit 7ef7e844883b23107c0b67e5cdcaf3947df7d856.

#### PR validation:

Run two typical workflows for Run-3 and Phase-2 with

```
runTheMatrix.py -l 11634.0,20834.0 -t 4 -j 8 --ibeos
```

And tested the event content of `TkAlMinBias` indeed changes as expected.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport, but to be backported down to 12.5.X in order to be used for the production happening in that cycle for L1T studies.

### CODE REVIEW

**Comment by mmusich** [comment]
FYI: @consuegs @antoniovagnerini

**Comment by mmusich** [comment]
type trk

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
@cmsbuild, please test

File: Alignment/CommonAlignmentProducer/python/ALCARECOTkAlBeamHalo_Output_cff.py [hunk 1]
```diff
@@ -6,13 +6,37 @@
...
+        'keep SiStripClusteredmNewDetSetVector_ALCARECOTkAlBeamHalo_*_*',
         'keep L1AcceptBunchCrossings_*_*_*',
         'keep L1GlobalTriggerReadoutRecord_gtDigis_*_*',
         'keep *_TriggerResults_*_*',
         'keep DcsStatuss_scalersRawToDigi_*_*')
 )
 
 import copy
+# in Run3, SCAL digis replaced by onlineMetaDataDigis
+import copy
```
Comment (tvami) [inline Alignment/CommonAlignmentProducer/python/ALCARECOTkAlBeamHalo_Output_cff.py:22]: it seems `import copy` is there twice, not sure if it matters...

File: Alignment/CommonAlignmentProducer/python/ALCARECOTkAlUpsilonMuMuPA_Output_cff.py [hunk 2]
```diff
@@ -15,5 +19,25 @@
 )
 
 import copy
+# in Run3, SCAL digis replaced by onlineMetaDataDigis
+import copy
```
Comment (tvami) [inline Alignment/CommonAlignmentProducer/python/ALCARECOTkAlUpsilonMuMuPA_Output_cff.py:23]: same here

Comment (mmusich) [reply Alignment/CommonAlignmentProducer/python/ALCARECOTkAlUpsilonMuMuPA_Output_cff.py:22]: I don't think it is really has any effect, but anyway I've removed the duplication in the last push.

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
@cmsbuild, please test
   * let's try again

**Comment by mandrenguyen** [comment]
+reconstruction
No change to reco

**Comment by mmusich** [comment]
@cms-sw/alca-l2 do you have any objections?
I'd like to backport this for the phase-2 production happening in 12_5_X.

**Comment by tvami** [comment]
+alca

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
@mmusich the situation with the IGs was a bit messy over the weekend. Now that it settled dowm, we remain with a crash in the Phase2 wf 20834.9 for the AlignmentTrackSelectorModule:ALCARECOMuAlOverlapsGeneralTracks module. It started with CMSSW_13_0_X_2022-12-17-1100, and I suspect it could come from the changes that were implemented with this PR: could you please have a look?

**Comment by mmusich** [comment]
@perrotta 

> I suspect it could come from the changes that were implemented with this PR

yes, it definitely comes from this PR.
A very simple fix (which is actually a workaround) is to comment this line:

https://github.com/cms-sw/cmssw/blob/9318eb0ff5982d62e93f385ffbe72e04887c93c8/CommonTools/RecoAlgos/src/ClusterStorer.cc#L53

This will solve the failure in the workflow, but then the AlCa step for the `vectorHit`s workflows still won't do the right thing (at least it is is not worse than it was before this PR). I am working on a more refined fix, but I cannot guarantee it will come before the EOY break. Let me know if this is acceptable from your point of view. From TRK point of view since VH are not the default this would still meet the goal of this PR.

**Comment by perrotta** [comment]
> nevermind, see #40363

Thank you @mmusich !

**Pull request merged**